tests/test_model.py

```diff
@@ -924,7 +926,50 @@ def test_hybrid_model_stages_progression(train_val_implicit_data, movielens_meta
 
     model.advance_stage()
 
-    assert model.hparams.stage == 'metadata_only'
+    assert model.hparams.stage == 'item_metadata_only'
+
+    model.advance_stage()
+
+    assert model.hparams.stage == 'all'
+
+    with pytest.raises(ValueError):
+        model.advance_stage()
+
+    with pytest.raises(ValueError):
+        model.set_stage('invalid_stage_name')
+
+    model = HybridModel(train=train, val=val, user_metadata=user_metadata_df)
+
+    assert model.hparams.stage == 'matrix_factorization'
+
+    model.advance_stage()
+
+    assert model.hparams.stage == 'user_metadata_only'
+
+    model.advance_stage()
+
+    assert model.hparams.stage == 'all'
+
+    with pytest.raises(ValueError):
+        model.advance_stage()
+
+    with pytest.raises(ValueError):
+        model.set_stage('invalid_stage_name')
+
+    model = HybridModel(train=train,
+                        val=val,
+                        item_metadata=movielens_metadata_df,
+                        user_metadata=user_metadata_df)
+
+    assert model.hparams.stage == 'matrix_factorization'
+
+    model.advance_stage()
+
+    assert model.hparams.stage == 'item_metadata_only'
+
+    model.advance_stage()
+
+    assert model.hparams.stage == 'user_metadata_only'
```

Comment:
[Optional] Should we break this up into an additional two tests? Generally, it's best to have one unit test that tests a single thing, that way if it breaks towards the beginning, we don't fix it only to realize that a later part of the test is also broken. What do you think?

Reply:
👍